src/main/java/qna/domain/Question.java

```diff
@@ -75,17 +55,46 @@ public boolean isOwner(User loginUser) {
         return writer.equals(loginUser);
     }
 
-    public Question setDeleted(boolean deleted) {
-        this.deleted = deleted;
-        return this;
-    }
-
     public boolean isDeleted() {
         return deleted;
     }
 
-    public List<Answer> getAnswers() {
-        return answers;
+    public List<DeleteHistory> delete(User loginUser) throws CannotDeleteException {
+        this.deleteInvalid(loginUser);
+
+        this.deleted = true;
+        List<DeleteHistory> deleteHistories = new ArrayList<>();
+        deleteHistories.add(createDeleteHistory());
+        deleteHistories.addAll(this.deleteAnswers());
+        return deleteHistories;
+    }
+
+    public DeleteHistory createDeleteHistory() {
+        return new DeleteHistory(ContentType.QUESTION, this.id, this.writer, LocalDateTime.now());
+    }
+
+    public void deleteInvalid(User loginUser) throws CannotDeleteException {
+        if (!this.isOwner(loginUser)) {
+            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
+        }
+
+        if (this.existsAnotherUserAnswer()) {
+            throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
+        }
+    }
+
+    public boolean existsAnotherUserAnswer() {
+        long count = this.answers.stream()
+                .filter(answer -> !answer.isOwner(this.writer))
+                .count();
+        return count > 0 ? true : false;
```

Comment:
return 코드 결과값이 boolean 타입이네요! 아래와 같이 작성 가능합니다.
```suggestion
        return count > 0;
```